pkg/monitor/cluster/networkmtu_test.go

```diff
@@ -0,0 +1,121 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"testing"
+
+	"go.uber.org/mock/gomock"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configfake "github.com/openshift/client-go/config/clientset/versioned/fake"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_metrics "github.com/Azure/ARO-RP/pkg/util/mocks/metrics"
+	testlog "github.com/Azure/ARO-RP/test/util/log"
+)
+
+func TestEmitNetworkMTU(t *testing.T) {
+	ctx := context.Background()
+
+	tests := []struct {
+		name           string
+		oc             *api.OpenShiftCluster
+		networkConfig  *configv1.Network
+		expectedMetric metricExpectation
+	}{
+		{
+			name: "MTU 1500 cluster with OpenShiftSDN",
+			oc: &api.OpenShiftCluster{
+				Properties: api.OpenShiftClusterProperties{
+					NetworkProfile: api.NetworkProfile{
+						MTUSize: api.MTU1500,
+					},
+				},
+			},
+			networkConfig: &configv1.Network{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: configv1.NetworkSpec{
+					NetworkType: "OpenShiftSDN",
+				},
+				Status: configv1.NetworkStatus{
+					ClusterNetworkMTU: 1500,
+				},
+			},
+			expectedMetric: metricExpectation{
+				name:  "network.mtu",
+				value: 1,
+				labels: map[string]string{
+					"mtu":          "1500",
+					"network_type": "OpenShiftSDN",
+				},
+			},
+		},
+		{
+			name: "MTU 1400 cluster with OVNKubernetes",
```

Comment:
It's not clear to me what value this second test adds after the first one. Do we expect OVN to have some difference to the outcome?

Reply:
Agreed with comment, which simplified and focused, single test case that tests the core functionality.